### PR TITLE
Pv2 2118 collate data

### PR DESCRIPTION
**Pull request by carlbm**
No acceptance tests yet - they are coming in a different task

### CODE REVIEW

**Review by hardik-desai-kcl — CHANGES_REQUESTED** [review]

File: src/MatchedLearnerApi.Application/Models/DatalockEvent.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+        // "Training" info
+        public string Reference { get; set; }
+        public int ProgrammeType { get; set; }
+        public int StandardCode { get; set; }
+        public int FrameworkCode { get; set; }
+        public int PathwayCode { get; set; }
+        public string FundingLineType { get; set; }
+        //public DateTime StartDate { get; set; }
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/Models/DatalockEvent.cs:28]: ✔️

File: src/MatchedLearnerApi.Application/Mappers/MatchedLearnerResultMapper.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                            ApprenticeshipId = payablePeriod.ApprenticeshipId,
+                            ApprenticeshipEmployerType = payablePeriod.ApprenticeshipEmployerType,
+                            TransferSenderAccountId = payablePeriod.Apprenticeship.TransferSendingEmployerAccountId
+                        }).Union(priceEpisode.NonPayablePeriods.SelectMany(nonPayablePeriod => nonPayablePeriod.Failures.GroupBy(failure => new PeriodDto
+                        {
+                            Period = nonPayablePeriod.Period,
+                            IsPayable = false,
+                            AccountId = failure.Apprenticeship.AccountId,
+                            ApprenticeshipId = failure.ApprenticeshipId,
+                            ApprenticeshipEmployerType = failure.Apprenticeship.ApprenticeshipEmployerType,
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/Mappers/MatchedLearnerResultMapper.cs:59]: This has been changed in the acceptance tests branch

File: src/MatchedLearnerApi.Application/Mappers/MatchedLearnerResultMapper.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                        NumberOfInstalments = priceEpisode.NumberOfInstalments,
+                        InstalmentAmount = priceEpisode.InstalmentAmount,
+                        CompletionAmount = priceEpisode.CompletionAmount,
+                        Periods = priceEpisode.PayablePeriods.Select(payablePeriod => new PeriodDto
+                        {
+                            Period = payablePeriod.Period,
+                            IsPayable = true,
+                            AccountId = payablePeriod.Apprenticeship.AccountId,
+                            ApprenticeshipId = payablePeriod.ApprenticeshipId,
+                            ApprenticeshipEmployerType = payablePeriod.ApprenticeshipEmployerType,
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/Mappers/MatchedLearnerResultMapper.cs:51]: This has been changed in the acceptance tests branch

File: src/MatchedLearnerApi.Application/PaymentsContext.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.HasDefaultSchema("Payments2");
+
+            modelBuilder.ApplyConfiguration(new DatalockEventConfiguration());
+            modelBuilder.ApplyConfiguration(new DatalockEventNonPayablePeriodConfiguration());
+            modelBuilder.ApplyConfiguration(new DatalockEventNonPayablePeriodFailureConfiguration());
+            modelBuilder.ApplyConfiguration(new DatalockEventPayablePeriodConfiguration());
+            modelBuilder.ApplyConfiguration(new DatalockEventPriceEpisodeConfiguration());
+
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/PaymentsContext.cs:34]: Not required as it follows the convention

File: src/MatchedLearnerApi.Types/TrainingDto.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    public class TrainingDto
+    {
+        public string Reference { get; set; }
+        public int ProgrammeType { get; set; }
+        public int StandardCode { get; set; }
+        public int FrameworkCode { get; set; }
+        public int PathwayCode { get; set; }
+        public string FundingLineType { get; set; }
+        public DateTime StartDate { get; set; }
+        public string Status { get; set; } //todo not used - why?
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Types/TrainingDto.cs:15]: ✔️

File: src/MatchedLearnerApi.Application/Mappers/MatchedLearnerResultMapper.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                    PriceEpisodes = dataLockEvent.PriceEpisodes.Select(priceEpisode => new PriceEpisodeDto
+                    {
+                        Identifier = priceEpisode.Identifier,
+                        AgreedPrice = priceEpisode.AgreedPrice,
+                        StartDate = priceEpisode.StartDate,
+                        EndDate = priceEpisode.ActualEndDate,
+                        NumberOfInstalments = priceEpisode.NumberOfInstalments,
+                        InstalmentAmount = priceEpisode.InstalmentAmount,
+                        CompletionAmount = priceEpisode.CompletionAmount,
+                        Periods = priceEpisode.PayablePeriods.Select(payablePeriod => new PeriodDto
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/Mappers/MatchedLearnerResultMapper.cs:45]: I've changed this to split it out into smaller functions

File: src/MatchedLearnerApi.Application/PaymentsContext.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace MatchedLearnerApi.Application
+{
+    public class PaymentsContext : DbContext, IPaymentsContext
+    {
+        private readonly string _connectionString;
+
+        public PaymentsContext(IConfiguration configuration)
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/PaymentsContext.cs:13]: ✔️

File: src/MatchedLearnerApi.Application/Repositories/EmployerIncentivesRepository.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        }
+
+        public async Task<MatchedLearnerResultDto> MatchedLearner(long ukprn, long uln)
+        {
+            var datalockEvent = await _context.DatalockEvents
+                .Include(x => x.PriceEpisodes).ThenInclude(x => x.NonPayablePeriods).ThenInclude(x => x.Failures).ThenInclude(x => x.Apprenticeship)
+                .Include(x => x.PriceEpisodes).ThenInclude(x => x.PayablePeriods).ThenInclude(x => x.Apprenticeship)
+                .Where(x => x.Ukprn == ukprn && x.Uln == uln)
+                .Where(x => x.AcademicYear == _context.DatalockEvents
+                    .Where(y => y.Ukprn == ukprn && y.Uln == uln).OrderByDescending(y => y.AcademicYear).ThenByDescending(y => y.IlrSubmissionWindowPeriod).Select(y => y.AcademicYear).FirstOrDefault())
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi.Application/Repositories/EmployerIncentivesRepository.cs:28]: ✔️

File: src/MatchedLearnerApi/IoC/MatchedLearnerApiRegistry.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+﻿using MatchedLearnerApi.Application;
+using MatchedLearnerApi.Application.Mappers;
+using MatchedLearnerApi.Application.Repositories;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace MatchedLearnerApi.IoC
+{
+    public class MatchedLearnerApiRegistry
+    {
+        public void Register(IServiceCollection services)
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/MatchedLearnerApi/IoC/MatchedLearnerApiRegistry.cs:10]: ✔️

**Comment by carlbm** [comment]
Superceded

**Pull request closed without merging**